routes/web/project/location/postgres.rb

```diff
@@ -47,6 +47,25 @@ class CloverWeb
 
         r.redirect "#{@project.path}#{st.subject.path}"
       end
+
+      r.post "reset_superuser_password" do
+        Authorization.authorize(@current_user.id, "Postgres:create", @project.id)
+        Authorization.authorize(@current_user.id, "Postgres:view", pg.id)
+
+        unless pg.server.primary?
+          response.status = 400
+          return {message: "Superuser password cannot be updated during restore!"}.to_json
+        end
```

Comment:
nit: It returns a JSON response. 
```suggestion
        unless pg.server.primary?
          flash["error"] = "Superuser password cannot be updated during restore!"
          return redirect_back_with_inputs
        end
```